decorator.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'nameable'
+
+class Decorator
+  include Nameable
+
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0...10]
```

Comment:
- [ ] Same here you should use `@nameable.correct_name[0...10]` to trim the name, kindly consider adjusting your code to not use the `super.[0...10]`